crates/telio-wg/src/wg.rs

```diff
@@ -677,9 +688,23 @@ impl State {
                     )
                     .await?;
                 }
+
+                if self.should_emit_downgrade_events
+                    && link_detection_update_result.should_notify
+                    && link_detection_update_result.link_state == Some(LinkState::Down)
+                {
+                    downgrade_event.to_be_downgraded.push(*key);
```

Comment:
I am a bit anxious about this list. The problem with it is that it lives a separate live where it's being filled with keys but I never see it being `poped` or `cleared`. This might introduce problems or fluctuations down the road where this list is consulted when judging the downgrade possibility. However the list needs to be carefully managed as it's a liability.

I have an idea - would it be possible to spin a tokio task in the wg_controller basically listening for the events additionally to the current logic? That way instead of saving the the list, the value would be pused down the channel and wg_controller would catch it and break the direct connection immediately. This way there would be no need for bookkeeping. How does that sound?

Reply:
I don't understand what you mean. This is exactly what I am doing. I build the list here, is sent through the channel, received in src/device/mod.rs, saved into requested_state, then I call wg_consolidate and then I clear the list. So there are no leftovers after the operation.